camayoc/tests/rho/test_scan.py

```diff
@@ -58,4 +67,77 @@ def test_scan(isolated_filesystem):
     rho_scan.logfile.close()
     rho_scan.close()
     assert rho_scan.exitstatus == 0, logfile
-    assert os.path.isfile(reportfile)
+    assert os.path.isfile(reportfile), logfile
+    with open(reportfile) as f:
+        fieldnames = csv.DictReader(f).fieldnames
+    assert sorted(fieldnames) == sorted(RHO_CONNECTION_FACTS + RHO_RHEL_FACTS)
+
+
+@pytest.mark.parametrize('facts', ('all', 'default', 'jboss', 'rhel', 'file'))
```

Comment:
You might be able to eschew some of the logic below if these were tuples or something that paired the 'facts' with the expected facts. That is probably overkill though.

Reply:
I preferred to add that logic in the test body to avoid having a big parametrize call and to be able to have a random set of facts when passing a file as value.